apps/web/src/components/Auth/RegisterForm.tsx

```diff
@@ -0,0 +1,205 @@
+import { Form, Controller } from "@litespace/ui/Form";
+import { Button } from "@litespace/ui/Button";
+import { useToast } from "@litespace/ui/Toast";
+import { useFormatMessage } from "@litespace/ui/hooks/intl";
+import React, { useCallback, useEffect, useMemo } from "react";
+import { useForm } from "react-hook-form";
+import { Link, useNavigate, useParams } from "react-router-dom";
+import { Route } from "@/types/routes";
+import { IUser } from "@litespace/types";
+import { useRegisterUser } from "@litespace/headless/user";
+import { useUserContext } from "@litespace/headless/context/user";
+import { Typography } from "@litespace/ui/Typography";
+import {
+  useValidateEmail,
+  useValidatePassword,
+} from "@litespace/ui/hooks/validation";
+import { useGoogle } from "@/hooks/google";
+import Google from "@litespace/assets/Google";
+import { getErrorMessageId } from "@litespace/ui/errorMessage";
+interface IForm {
+  email: string;
+  password: string;
+  confirmedPassword: string;
+}
+
+type Role = (typeof roles)[number];
+
+const roles = [IUser.Role.Tutor, IUser.Role.Student] as const;
+
+const origin = location.origin;
+const callbackUrl = origin.concat(Route.VerifyEmail);
+
+const RegisterForm: React.FC = () => {
+  const intl = useFormatMessage();
+  const user = useUserContext();
+  const navigate = useNavigate();
+  const toast = useToast();
+  const { role } = useParams<{ role: Role }>();
+  const { watch, handleSubmit, control, formState } = useForm<IForm>({
+    defaultValues: {
+      email: "",
+      password: "",
+      confirmedPassword: "",
+    },
+  });
+  const errors = formState.errors;
+
+  const validateEmail = useValidateEmail(true);
+  const validatePassword = useValidatePassword(true);
+  const isValidRole = useMemo(() => role && roles.includes(role), [role]);
+  const google = useGoogle({ role: isValidRole ? role : undefined });
+  const email = watch("email");
+  const password = watch("password");
+  const confirmedPassword = watch("confirmedPassword");
+
+  useEffect(() => {
+    if (!isValidRole) return navigate(Route.Root);
+  }, [isValidRole, navigate]);
+
+  const onSuccess = useCallback(
+    async ({ user: info, token }: IUser.RegisterApiResponse) => {
+      user.set({ user: info, token });
+      navigate(Route.Root);
+    },
+    [user, navigate]
+  );
+
+  const onError = useCallback(
+    (error: unknown) => {
+      const errorMessage = getErrorMessageId(error);
+      toast.error({
+        title: intl("register.error"),
+        description: intl(errorMessage),
+      });
+    },
+    [intl, toast]
+  );
+
+  const mutation = useRegisterUser({ onSuccess, onError });
+
+  const onSubmit = useMemo(
+    () =>
+      handleSubmit((payload: IForm) => {
+        if (!isValidRole || !role) return;
+        return mutation.mutate({
+          ...payload,
```

Comment:
This is not valid. The payload has a prop called `confirmedPassword` and there is no need to pass it to the mutation.